src/java/org/apache/cassandra/index/sai/disk/vector/CassandraOnHeapGraph.java

```diff
@@ -297,15 +298,66 @@ public SegmentMetadata.ComponentMetadataMap writeData(IndexDescriptor indexDescr
                 if (vectorPostings.shouldAppendDeletedOrdinal())
                     deletedOrdinals.add(vectorPostings.getOrdinal());
             }
+
+            // map of existing ordinal to rowId (aka new ordinal if remapping is possible)
+            BiMap <Integer, Integer> ordinalMap = HashBiMap.create();
+            boolean canFastFindRows = false;
+            if (deletedOrdinals.size() == 0) {
+                canFastFindRows = true;
+                int minRow = Integer.MAX_VALUE;
+                int maxRow = Integer.MIN_VALUE;
+                for (VectorPostings<T> vectorPostings : postingsMap.values())
+                {
+                    if (vectorPostings.getRowIds().size() != 1)
+                    {
+                        canFastFindRows = false;
+                        ordinalMap.clear();
+                        break;
+                    }
+                    int rowId = vectorPostings.getRowIds().getInt(0);
+                    int ordinal = vectorPostings.getOrdinal();
+                    minRow = Math.min(minRow, rowId);
+                    maxRow = Math.max(maxRow, rowId);
+                    if(rowId != ordinal)
```

Comment:
i don't think it's worth special casing the rare condition of `rowId == ordinal`, just add a map entry for everything